File: frontend/src/api/calls.ts
import { api } from '@/lib/api';

export interface Call {
  id: string;
  call_uid?: string;
  feed_id?: number;
  tg_id?: number;
  started_at?: string;
  duration_ms?: number;
  processed?: boolean;
}

export async function getRecentCalls(limit: number = 50): Promise<Call[]> {
  try {
    const res = await api.get('/calls', { params: { limit } });
    return res.data ?? [];
  } catch (e) {
    console.error('getRecentCalls failed', e);
    return [];
  }
}

export async function getCallsByFeed(feedId: number, limit: number = 50): Promise<Call[]> {
  try {
    const res = await api.get('/calls', { params: { feed_id: feedId, limit } });
    return res.data ?? [];
  } catch (e) {
    console.error('getCallsByFeed failed', e);
    return [];
  }
}

// Export alias for Dashboard compatibility
export const getCalls = getRecentCalls;

File: app_api/models/calls.py
from datetime import datetime

from pydantic import BaseModel


class CallMetadata(BaseModel):
    """Call metadata response model."""
    call_uid: str
    feed_id: int | None = None
    tg_id: int | None = None
    tag_id: int | None = None
    url: str | None = None
    started_at: datetime | None = None
    ended_at: datetime | None = None
    duration_ms: int | None = None
    size_bytes: int | None = None
    fetched_at: datetime | None = None
    processed: bool = False
    error: str | None = None

    # Transformed fields (added by API transformer for frontend compatibility)
    timestamp: str | None = None

    class Config:
        from_attributes = True


class CallDetail(CallMetadata):
    """Extended call detail with transcript info."""
    transcript_text: str | None = None
    transcript_confidence: float | None = None
    audio_url: str | None = None


class HourlyStats(BaseModel):
    """Hourly call statistics."""
    hour: datetime
    count: int


class FeedStats(BaseModel):
    """Statistics grouped by feed."""
    feed_id: int
    count: int
    avg_duration_ms: float | None = None
